includes/Modules/Analytics.php

```diff
@@ -937,115 +943,114 @@ protected function parse_data_response( $method, $datapoint, $response ) {
 					// TODO: Parse this response to a regular array.
 					break;
 				case 'accounts-properties-profiles':
-					$response = array(
-						// TODO: Parse this response to a regular array.
-						'accounts'   => $response->getItems(),
+					/* @var \Google_Service_Analytics_Accounts $response listManagementAccounts response. */
+					$accounts            = (array) $response->getItems();
+					$account_ids         = array_map(
+						function ( \Google_Service_Analytics_Account $account ) {
+							return $account->getId();
+						},
+						$accounts
+					);
+					$properties_profiles = array(
 						'properties' => array(),
 						'profiles'   => array(),
 					);
+					$existing_tag        = $this->get_existing_tag();
 
-					$found_account_id = false;
-					$matched_property = false;
-					$existing_tag     = $this->_existing_tag_account;
-
-					$this->_existing_tag_account = null; // Set back to null.
-
-					if ( empty( $existing_tag ) ) {
+					if ( $existing_tag['accountId'] ) {
```

Comment:
This should also check if there is an `existingPropertyId` passed, and then pass that on to `properties-profiles`, alongside with the `existingAccountId` as `accountId`.

Reply:
As it is, `$existing_tag` will have both set if one of them is set, although we could add the additional check if it would be possible for only one to be set?

Currently both `accountId` and `propertyId` are being passed to `properties-profiles` in this case.

Reply:
This ties in with my other comment from above, I think we should treat them independently. Most importantly though, we need to pass `existingAccountId` as simply `accountId`.